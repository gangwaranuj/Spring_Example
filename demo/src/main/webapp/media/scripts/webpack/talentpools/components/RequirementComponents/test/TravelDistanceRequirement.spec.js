import React from 'react';
import { shallow } from 'enzyme';
import { WMRaisedButton } from '@workmarket/front-end-components';
import TravelDistanceRequirement from '../TravelDistanceRequirement';
import { initialState } from '../../../reducers';

const dummyFunc = jest.fn();
const requirementComponentData = initialState.get('requirementsData').toJS();

describe('<TravelDistanceRequirement />', () => {
	describe('Rendering consistency', () => {
		it('should render the same way every time with all props', () => {
			const snapshot = shallow(
				<TravelDistanceRequirement.WrappedComponent
					sliderChange={ dummyFunc }
					applyRequirement={ dummyFunc }
					onGoogleAPILoaded={ dummyFunc }
					onChangeGoogleAddress={ dummyFunc }
					requirementComponentData={ requirementComponentData }
				/>,
			);
			expect(snapshot).toMatchSnapshot();
		});
	});

	describe('Interaction', () => {
		let wrapper;
		let button;

		const shallowRenderComponent = () => shallow(
			<TravelDistanceRequirement.WrappedComponent
				sliderChange={ dummyFunc }
				applyRequirement={ dummyFunc }
				onGoogleAPILoaded={ dummyFunc }
				onChangeGoogleAddress={ dummyFunc }
				requirementComponentData={ requirementComponentData }
			/>
		);

		beforeEach(() => {
			wrapper = shallowRenderComponent();
			button = wrapper.find(WMRaisedButton);
		});

		it('should have a <WMRaisedButton /> add button set to disabled', () => {
			const addButton = button.get(0);
			expect(addButton.props.disabled).toBeTruthy();
		});
	});
});

