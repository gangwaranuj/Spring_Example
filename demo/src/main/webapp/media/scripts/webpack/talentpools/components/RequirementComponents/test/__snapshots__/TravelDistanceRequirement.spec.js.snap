// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TravelDistanceRequirement /> Rendering consistency should render the same way every time with all props 1`] = `
<WMFormRow
  data-component-identifier="requirements_row"
  id="requirements-abandoned"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <div>
      Distance
    </div>
    <div
      style={
        Object {
          "alignSelf": "flex-end",
        }
      }
    >
      10
       miles
    </div>
    <div
      style={
        Object {
          "margin": "-1em 0",
        }
      }
    >
      <WMSlider
        data-component-identifier="requirements_slider"
        max={100}
        min={10}
        name="maxTravelDistanceValue"
        onChange={[Function]}
        step={1}
        value={10}
      />
    </div>
    <WMGoogleLocationInput
      changeGoogleAddress={[Function]}
      data-component-identifier="requirements__travelDistanceLocation"
      googleInitialized={false}
      onGoogleAPILoaded={[Function]}
      value=", , null, "
    />
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row-reverse",
        }
      }
    >
      <WMRaisedButton
        data-component-identifier="requirements_button"
        disabled={true}
        label="Add this requirement"
        onClick={[Function]}
        secondary={true}
        style={
          Object {
            "margin": "1em 0 0 1em",
          }
        }
      />
    </div>
  </div>
</WMFormRow>
`;
